app/models/session.rb

```diff
@@ -39,6 +40,8 @@ def ttl_in_time
     Time.current.utc + ttl
   end
 
+  def authenticated_by_ssoe
+    @ssoe_transaction != nil
```

Comment:
Did the method get truncated (e.g. where is the `end`)?

Reply:
yeah, must have saved the buffer too early.  I'll fix in a few